### PR TITLE
MMRazor Prune - Object Detection

### PR DESCRIPTION
**Pull request by Akash-guna**
Previous PR #10

### CODE REVIEW

File: prune/mmrazorprune/main.py [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+            input_shape=input_shape,
+        ),
+        save_ckpt_thr=[target_flop_ratio],
+    ),
+]
+
+        """
+        with open("current_fisher_config.py", "w") as f:
+            f.write(config)
+        return
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:128]: confirm file creation is either cleaned post job completion or created within job dir or cache dir

File: prune/mmrazorprune/main.py [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+optim_wrapper = dict(optimizer=dict(lr=finetune_lr))
+
+# remove pruning related hooks
+custom_hooks = _base_.custom_hooks[:-2]
+
+# delete ddp
+model_wrapper_cfg = None'''
+        with open("current_fisher_finetune_config.py", "w") as f:
+            f.write(config)
+        return
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:165]: same as above.

File: prune/mmrazorprune/main.py [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+
+    def customize_config(self):
+        self.write_config()
+        cfg = Config.fromfile("current_fisher_config.py")
+        cfg["train_cfg"]["max_epochs"] = self.prune_epochs
+        cfg.work_dir = self.job_path
+        cfg.dump("current_config_new.py")
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:172]: as above. and subsequently all other such occurrences

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

File: prune/mmrazorprune/main.py (hunk 1 above)
Comment (Akash-guna) [reply prune/mmrazorprune/main.py]: Done

File: prune/mmrazorprune/main.py (hunk 2 above)
Comment (Akash-guna) [reply prune/mmrazorprune/main.py]: Rectified

File: prune/mmrazorprune/main.py (hunk 3 above)
Comment (Akash-guna) [reply prune/mmrazorprune/main.py:172]: Everything Rectified

File: prune/mmrazorprune/main.py [hunk 4]
```diff
@@ -221,8 +221,8 @@ def compress_model(self):
         cfg = self.customize_config()
         command = [
             "python",
-            "/mmrazor/tools/train.py",
-            "current_config_new.py",
+            "vision/core/utils/mmrazortrain.py",
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:224]: lets derive this path at runtime.

File: prune/mmrazorprune/main.py [hunk 5]
```diff
@@ -221,8 +221,8 @@ def compress_model(self):
...
+            f"{self.cache_path}current_config_new.py",
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:225]: I believe there should be a forward slash (/) after self.cache_path

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

File: prune/mmrazorprune/main.py [hunk 6]
```diff
@@ -0,0 +1,239 @@
...
+        with open(f"{self.cache_path}/current_fisher_finetune_config.py", "w") as f:
+            f.write(config)
+        return
+
+    def customize_config(self):
+        self.write_config()
+        cfg = Config.fromfile(f"{self.cache_path}/current_fisher_config.py")
+        cfg["train_cfg"]["max_epochs"] = self.prune_epochs
+        cfg.work_dir = self.job_path
+        cfg.dump("current_config_new.py")
```
Comment (shwu-nyunai) [inline prune/mmrazorprune/main.py:173]: we are dumping this cfg, can't confirm if this is inside cache path? lets do a second check. if safe, kindly resolve.

**Review by shwu-nyunai — CHANGES_REQUESTED** [review]

**Review by shwu-nyunai — APPROVED** [review]

**Review by Arnav0400 — APPROVED** [review]

**Pull request merged**